be/src/http/download_action.cpp

```diff
@@ -62,7 +63,7 @@ DownloadAction::DownloadAction(ExecEnv* exec_env, const std::string& error_log_r
                   "canonicalize path " + error_log_root_dir + " failed");
 }
 
-void DownloadAction::handle_normal(HttpRequest* req, const std::string& file_param) {
+void DownloadAction::handle_normal(HttpRequest* req, const std::string& file_param, const std::string& type) {
```

Comment:
```suggestion
void DownloadAction::handle_normal(HttpRequest* req, const std::string& file_param) {
```
It seems that you can get the type parameter from `req`.